src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,336 @@
+                    .url(requestUrl)
+                    .post(body)
+                    .header("Content-Type", "application/json")
+                    .header("Authorization", String.format("Bearer %s", bearerToken))
+                    .build();
+            call = client.newCall(request);
+            Response response = call.execute();
+            String responseBody = response.body().string();
+            LOG.info(mapper.readTree(responseBody).get("state"));
+            waitForStateOfCluster(clusterId,"RUNNING", TimeUnit.MINUTES.toMillis(timeoutForClusterStateWait));
+        } catch (Exception e) {
+            throw new CloudException(String.format("Set member size of cluster with id %s is failed, Rc stack trace is: %s", clusterId, e.getStackTrace()));
+        }
+    }
+
+    public CloudCluster getHazelcastCloudCluster(String clusterId) throws CloudException {
+        try {
+            String query = String.format("{\"query\": \"query { cluster(clusterId: \\\"%s\\\") { id name hazelcastVersion isTlsEnabled state discoveryTokens {source,token}}}\"}", clusterId);
+            String responseBody = prepareAndSendRequest(query).body().string();
+            LOG.info(maskValueOfToken(responseBody));
+            JsonNode rootNode = mapper.readTree(responseBody).get("data").get("cluster");
+            if(rootNode.asText().equalsIgnoreCase("null"))
+                return null;
+
+            CloudCluster cluster = new CloudCluster(rootNode.get("id").asText(), rootNode.get("name").asText(), getConnectionName(rootNode.get("id").asText()), rootNode.get("hazelcastVersion").asText(), rootNode.get("isTlsEnabled").asBoolean(), rootNode.get("state").asText(), rootNode.get("discoveryTokens").elements().next().get("token").asText(), null, null);
+            if(rootNode.get("isTlsEnabled").asBoolean())
+            {
+                cluster.setCertificatePath(downloadCertificatesAndGetPath(cluster.getId()));
+                cluster.setTlsPassword(getTlsPassword(cluster.getId()));
+            }
+            return cluster;
+        } catch (Exception e) {
+            throw new CloudException(String.format("Get cluster with id %s is failed, Rc stack trace is: ", clusterId, e.getStackTrace()));
+        }
+    }
+
+    public CloudCluster stopHazelcastCloudCluster(String clusterId) throws CloudException {
+        try {
+            String query = String.format("{\"query\": \"mutation { stopCluster(clusterId:\\\"%s\\\") { clusterId }}\"}", clusterId);
+            String responseBody = prepareAndSendRequest(query).body().string();
+            LOG.info(maskValueOfToken(responseBody));
+            waitForStateOfCluster(clusterId, "STOPPED", TimeUnit.MINUTES.toMillis(timeoutForClusterStateWait));
+            return getHazelcastCloudCluster(clusterId);
+        } catch (Exception e) {
+            throw new CloudException(String.format("Stop cluster with id %s is failed, Rc stack trace is: %s", clusterId, e.getStackTrace()));
```

Comment:
```suggestion
            throw new CloudException(String.format("Stop cluster with id %s is failed, Rc stack trace is: %s", clusterId, Arrays.toString(e.getStackTrace())));
```

Please also correct other such exceptions, they have different problems as well (like missing %s format).